File: src/Haus.Web.Host/client-app/src/app/shared/models/lighting.model.ts
import {LightingState} from "./lighting-state.model";
import {LightingColorModel} from "./lighting-color.model";
import {LightingConstraintsModel} from "./lighting-constraints.model";

export interface LightingModel {
  state: LightingState;
  level: number;
  temperature: number;
  color: LightingColorModel;
  constraints: LightingConstraintsModel;
}

File: src/Haus.Core.Models/Lighting/LightingModel.cs
namespace Haus.Core.Models.Lighting;

public record LightingModel(
    LightingState State = LightingDefaults.State,
    LevelLightingModel? Level = null,
    TemperatureLightingModel? Temperature = null,
    ColorLightingModel? Color = null
)
{
    public LevelLightingModel Level { get; init; } = Level ?? new LevelLightingModel();
}
